KickIT@Eclipse/Doxygen/man/man3/VirtualKickerWindow.3
.TH "VirtualKickerWindow" 3 "Mon Sep 25 2017" "KickIT@Eclipse" \" -*- nroff -*-
.ad l
.nh
.SH NAME
VirtualKickerWindow \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <VirtualKickerWindow\&.hpp>\fP
.PP
Inherits QWidget\&.
.SS "Public Slots"

.in +1c
.ti -1c
.RI "void \fBnewBallStatus\fP (\fBBallStatus\fP bs)"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBVirtualKickerWindow\fP (bool showLines, bool \fBdrawBallPositionWithMouse\fP)"
.br
.ti -1c
.RI "void \fBpaintEvent\fP (QPaintEvent *event)"
.br
.ti -1c
.RI "void \fBmouseMoveEvent\fP (QMouseEvent *e)"
.br
.ti -1c
.RI "void \fBmouseReleaseEvent\fP (QMouseEvent *e)"
.br
.ti -1c
.RI "void \fBkeyPressEvent\fP (QKeyEvent *event)"
.br
.ti -1c
.RI "void \fBsetTableController\fP (\fBTableControllerInterface\fP *t)"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "int \fBWINDOW_SIZE_X\fP = 1024"
.br
.ti -1c
.RI "int \fBWINDOW_SIZE_Y\fP = 768"
.br
.ti -1c
.RI "double \fBSCALE\fP"
.br
.ti -1c
.RI "double \fBTABLE_MARGIN\fP"
.br
.ti -1c
.RI "double \fBTABLE_HEIGHT\fP"
.br
.ti -1c
.RI "double \fBTABLE_WIDTH\fP"
.br
.ti -1c
.RI "double \fBGOAL_SIZE\fP"
.br
.ti -1c
.RI "double \fBGOAL_TO_KEEPER\fP"
.br
.ti -1c
.RI "double \fBGOAL_TO_DEFENSE\fP"
.br
.ti -1c
.RI "double \fBBALLDIAMETER\fP"
.br
.ti -1c
.RI "QLabel * \fBspeedDisplay\fP"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "\fBTableConfig\fP \fBtconf\fP"
.br
.ti -1c
.RI "\fBTableControllerInterface\fP * \fBtc\fP"
.br
.ti -1c
.RI "QPoint \fBlastAdded\fP"
.br
.ti -1c
.RI "QPoint * \fBtopLeft\fP"
.br
.ti -1c
.RI "QLine * \fBkeeperBar\fP"
.br
.ti -1c
.RI "QPoint * \fBkeeper\fP"
.br
.ti -1c
.RI "QLine * \fBdefenseBar\fP"
.br
.ti -1c
.RI "QPoint * \fBdefense\fP [2]"
.br
.ti -1c
.RI "QLine * \fBballMovementVector\fP"
.br
.ti -1c
.RI "std::vector< QPoint * > \fBmouseTrail\fP"
.br
.ti -1c
.RI "QPoint * \fBball\fP"
.br
.ti -1c
.RI "bool \fBshowRows\fP"
.br
.ti -1c
.RI "bool \fBdrawBallPositionWithMouse\fP"
.br
.ti -1c
.RI "float \fBvMax\fP"
.br
.ti -1c
.RI "clock_t \fBlast_ball_update\fP"
.br
.in -1c
.SH "Constructor & Destructor Documentation"
.PP 
.SS "VirtualKickerWindow::VirtualKickerWindow (bool showLines, bool drawBallPositionWithMouse)"

.SH "Member Function Documentation"
.PP 
.SS "void VirtualKickerWindow::keyPressEvent (QKeyEvent * event)"

.SS "void VirtualKickerWindow::mouseMoveEvent (QMouseEvent * e)"

.SS "void VirtualKickerWindow::mouseReleaseEvent (QMouseEvent * e)"

.SS "void VirtualKickerWindow::newBallStatus (\fBBallStatus\fP bs)\fC [slot]\fP"

.SS "void VirtualKickerWindow::paintEvent (QPaintEvent * event)"

.SS "void VirtualKickerWindow::setTableController (\fBTableControllerInterface\fP * t)"

.SH "Member Data Documentation"
.PP 
.SS "QPoint* VirtualKickerWindow::ball\fC [protected]\fP"

.SS "double VirtualKickerWindow::BALLDIAMETER"

.SS "QLine* VirtualKickerWindow::ballMovementVector\fC [protected]\fP"

.SS "QPoint* VirtualKickerWindow::defense[2]\fC [protected]\fP"

.SS "QLine* VirtualKickerWindow::defenseBar\fC [protected]\fP"

.SS "bool VirtualKickerWindow::drawBallPositionWithMouse\fC [protected]\fP"

.SS "double VirtualKickerWindow::GOAL_SIZE"

.SS "double VirtualKickerWindow::GOAL_TO_DEFENSE"

.SS "double VirtualKickerWindow::GOAL_TO_KEEPER"

.SS "QPoint* VirtualKickerWindow::keeper\fC [protected]\fP"

.SS "QLine* VirtualKickerWindow::keeperBar\fC [protected]\fP"

.SS "clock_t VirtualKickerWindow::last_ball_update\fC [protected]\fP"

.SS "QPoint VirtualKickerWindow::lastAdded\fC [protected]\fP"

.SS "std::vector<QPoint*> VirtualKickerWindow::mouseTrail\fC [protected]\fP"

.SS "double VirtualKickerWindow::SCALE"

.SS "bool VirtualKickerWindow::showRows\fC [protected]\fP"

.SS "QLabel* VirtualKickerWindow::speedDisplay"

.SS "double VirtualKickerWindow::TABLE_HEIGHT"

.SS "double VirtualKickerWindow::TABLE_MARGIN"

.SS "double VirtualKickerWindow::TABLE_WIDTH"

.SS "\fBTableControllerInterface\fP* VirtualKickerWindow::tc\fC [protected]\fP"

.SS "\fBTableConfig\fP VirtualKickerWindow::tconf\fC [protected]\fP"

.SS "QPoint* VirtualKickerWindow::topLeft\fC [protected]\fP"

.SS "float VirtualKickerWindow::vMax\fC [protected]\fP"

.SS "int VirtualKickerWindow::WINDOW_SIZE_X = 1024"

.SS "int VirtualKickerWindow::WINDOW_SIZE_Y = 768"


.SH "Author"
.PP 
Generated automatically by Doxygen for KickIT@Eclipse from the source code\&.


KickIT_Ntb/VirtualKicker/VirtualKickerWindow.cpp



#include "VirtualKickerWindow.hpp"

VirtualKickerWindow::VirtualKickerWindow(BallTrackerMock* b) {
    
//VirtualKickerWindow::VirtualKickerWindow() {
    setMouseTracking(true);
    //this->btm = b;
    topLeft = new QPoint(TABLE_MARGIN,TABLE_MARGIN);
    bottomRight = new QPoint( WINDOW_SIZE_X - TABLE_MARGIN, WINDOW_SIZE_Y - TABLE_MARGIN);
    keeperBar = new QLine(topLeft->x()+20,topLeft->y()-10,topLeft->x()+20, bottomRight->y() +10);
    keeper = new QPoint(keeperBar->x1(), keeperBar->y1() + keeperBar->dy()/2);
    
}

void VirtualKickerWindow::mouseMoveEvent(QMouseEvent* e){
    btm->mouseMove(e);
    
    if( (abs(e->pos().x() -lastAdded.x()) + abs(e->pos().y() - lastAdded.y()))  > 50 && e->buttons() & Qt::LeftButton){
        mouseTrail.push_back(new QPoint(e->pos().x(), e->pos().y()));
        lastAdded.setX(e->pos().x());
        lastAdded.setY(e->pos().y());
    }
    repaint();
}

void VirtualKickerWindow::mouseReleaseEvent(QMouseEvent* e) {
    mouseTrail.clear();
}





void VirtualKickerWindow::paintEvent(QPaintEvent *event){
            
    //create a QPainter and pass a pointer to the device.
    //A paint device can be a QWidget, a QPixmap or a QImage
    QPainter painter(this);
    

    //a simple line
    //painter.drawLine(1,1,100,100);

    //create a black pen that has solid line
    //and the width is 2.
    QPen myPen(Qt::black, 2, Qt::SolidLine);
    painter.setPen(myPen);
    //painter.drawLine(100,100,100,1);

    //draw a point
    //myPen.setColor(Qt::red);
    //painter.drawPoint(110,110);

    //draw a polygon
    //QPolygon polygon;
    //polygon << QPoint(130, 140) << QPoint(180, 170)
    //         << QPoint(180, 140) << QPoint(220, 110)
    //         << QPoint(140, 100);
    //painter.drawPolygon(polygon);

    //draw an ellipse
    //The setRenderHint() call enables antialiasing, telling QPainter to use different
    //color intensities on the edges to reduce the visual distortion that normally
    //occurs when the edges of a shape are converted into pixels
    //painter.setRenderHint(QPainter::Antialiasing, true);
    //painter.setPen(QPen(Qt::black, 3, Qt::DashDotLine, Qt::RoundCap));
    //painter.setBrush(QBrush(Qt::green, Qt::SolidPattern));

    //painter.drawEllipse(200, 80, 400, 240);


    painter.drawRect(topLeft->x(),topLeft->y() , WINDOW_SIZE_X-2*TABLE_MARGIN, WINDOW_SIZE_Y - 2*TABLE_MARGIN);
    painter.drawLine(*keeperBar);
    painter.setPen(QPen(Qt::blue, 20, Qt::DashDotLine, Qt::RoundCap));
    painter.drawPoint(*keeper);
    
    for(long unsigned int i = 0; i<mouseTrail.size(); i++){
        painter.drawPoint(*mouseTrail[i]);
        //std::cout << "p";
    }

}